altanalyze3/components/merging/main_new.py

```diff
@@ -0,0 +1,75 @@
+from genericpath import exists
+from operator import truediv
+
+import os
+from functools import cache
+import logging
+
+class   JunctionAnnotation:
+    def __init__(self):
+        self.start_annotation = {},
+        self.stop_annotation = {},
+        self.junction_coordinate_BAM_dict = {}
+
+    def each_junction_annotation(self,junction_dir, gene_model_all):
+        '''
+        Input args: path to all BAM derived sample txt files, gene_model_all txt file(hg38)
+        Output: map of annotations for a given junction
+        '''
+        junction_files = os.listdir(junction_dir)
+        junction_dict = self.generate_junction_dict(gene_model_all)
+        junction_coordinate_BAM_listof_dicts = []
```

Comment:
I don't see it used anywhere

Reply:
this code is still in progress. I am closing this PR since I opened it for review (to review the older version of code) in our meeting from last week and the code has evolved since then.